controllers/jujucluster_controller.go

```diff
@@ -114,14 +116,19 @@ func (r *JujuClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	// Check if a secret containing cloud user data has been created yet
-	cloudSecret, err := r.getCloudSecret(ctx, jujuCluster)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-	if cloudSecret == nil {
-		log.Info(fmt.Sprintf("secret %s not found in namespace %s, please create it", jujuCluster.Name+"-cloud-secret", jujuCluster.Namespace))
-		return ctrl.Result{Requeue: true}, nil
+	var credentialSecret *kcore.Secret = nil
+	// If cloud has auth types, check for a credential secret before doing anything else
+	// A cloud with 0 auth types means credentials are not required
```

Comment:
We check authtypes of the cloud to determine if creds are needed, as some clouds may not need creds necessarily